core/src/tangram.cpp

```diff
@@ -241,6 +247,23 @@ void render() {
     FrameInfo::draw(*m_view, *m_tileManager);
 
     while (Error::hadGlError("Tangram::render()")) {}
+
+
+    bool ready = !(m_view->changedOnLastUpdate() ||
+                   m_tileManager->hasTileSetChanged() ||
+                   m_tileManager->hasLoadingTiles() ||
+                   m_labels->needUpdate());
+    if (ready) {
+        for (auto& ease : m_eases) {
+            if (!ease.finished()) { ready = false; }
+        }
+    }
+
+    m_renderDoneCallback(ready);
```

Comment:
@blair1618 How about this last commit? Just return true from render() whenever the view is fully loaded, keeping the listener thing completely in MapController